vortex-ffi/src/file.rs

```diff
@@ -1,29 +1,69 @@
 //! FFI interface for Vortex File I/O.
 
-use std::ffi::c_char;
+use std::ffi::{CStr, c_char, c_int};
+use std::str::FromStr;
 use std::sync::Arc;
 
-use futures::StreamExt;
+use object_store::aws::{AmazonS3Builder, AmazonS3ConfigKey};
+use object_store::azure::{AzureConfigKey, MicrosoftAzureBuilder};
+use object_store::gcp::{GoogleCloudStorageBuilder, GoogleConfigKey};
 use object_store::local::LocalFileSystem;
+use object_store::{ObjectStore, ObjectStoreScheme};
+use url::Url;
 use vortex::dtype::DType;
-use vortex::error::VortexExpect;
+use vortex::error::{VortexError, VortexExpect, VortexResult, vortex_bail};
+use vortex::expr::{Identity, select};
 use vortex::file::{GenericVortexFile, VortexFile, VortexOpenOptions};
 use vortex::io::ObjectStoreReadAt;
 
 use crate::stream::{FFIArrayStream, FFIArrayStreamInner};
-use crate::{RUNTIME, to_string};
+use crate::{RUNTIME, to_string, to_string_vec};
 
-#[repr(C)]
 pub struct FFIFile {
     pub(crate) inner: VortexFile<GenericVortexFile<ObjectStoreReadAt>>,
 }
 
+/// Options supplied for opening a file.
+#[repr(C)]
```

Comment:
this is not accessed opaquely, and thus needs to be repr(C) so we can build this struct in other languages

Reply:
perhaps this would be better opaque,  and then build it using a function over FFI that accepts all of these fields as parameters. but that kinda is just the same thing? idk